### ISSUE TITLE
Document how to create in-tree plugins

### ISSUE THREAD

**Issue by alculquicondor** [kubernetes/kubernetes#91120]
We have recently changed how plugin args are parsed, defaulted and (#90859) validated.

The process is more complicated than most standard API uses.
We need to document this.

/kind documentation

/help

**Comment by manishdangi98** [comment]
please assign me this one

**Comment by alculquicondor** [comment]
@manishdangi98 you can take it

**Comment by alculquicondor** [comment]
cc @adtac how's looking into modifying one plugin's configuration now

**Comment by pravarag** [comment]
@manishdangi98 are you still working on it? Otherwise I would like to work on this issue :) @alculquicondor

**Comment by Aut0R3V** [comment]
@adtac anything I can do here?

**Comment by adtac** [comment]
@Aut0R3V see https://github.com/kubernetes/kubernetes/tree/master/pkg/scheduler/framework/plugins#creating-a-new-in-tree-plugin - that section essentially needs to be written.

**Comment by Aut0R3V** [comment]
Thanks! Working on it.

**Comment by Aut0R3V** [comment]
> This is divided into:
> 
> * [x]   moving validation of all plugin args but Node Resources ones  #91241
> * [x]  moving validation of Node Resources plugin args #91446
> * [ ]  using moved validation in `ValidateKubeSchedulerConfiguration`,
>   and improving code and error message consistency
> * [ ]  moving validation of recently added plugin args

**Comment by Aut0R3V** [comment]
@adtac I went through the PR's mentioned in #90859 and the respective commits.

@pancernik 's explanation gave rough overview here. 
>   moving validation of all plugin args but Node Resources ones
>   moving validation of Node Resources plugin args 
>   using moved validation in `ValidateKubeSchedulerConfiguration`,
>   and improving code and error message consistency
>   moving validation of recently added plugin args

Basically what I have to rewrite the documentation in this context right? Please correct if I'm wrong here.

**Comment by coderfornow** [comment]
can I take this, is this still open?
@adtac

**Comment by Aut0R3V** [comment]
I was waiting for a response on my ideas here :)
I'm still interested in this, meanwhile it'll be great to hear your ideas on the issue!

**Comment by adtac** [comment]
>I went through the PR's mentioned in #90859 and the respective commits

I don't think that issue covers all the aspects of creating a new in-tree plugin.

>Basically what I have to rewrite the documentation in this context right?

I'm not sure what you mean by rewrite (the documentation for creating a new in-tree plugin doesn't exist right now), but yes, the PRs in that issue will be useful as context.

FWIW I think it might be best to leave this to people who have created in-tree plugins in the past (excludes me).

**Comment by Aut0R3V** [comment]
Sure okay 👍
